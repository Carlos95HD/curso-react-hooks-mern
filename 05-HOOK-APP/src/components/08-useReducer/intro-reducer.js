//Estado Inicial
const initialState = [{
  id: 1,
  todo:'Comprar pan',
  done: false
}];

//Definimos el reducer
const todoReducer = ( state = initialState, action ) => {

  if (action?.type === 'agregar') {
    return [ ...state, action.payload ]
  }

  return state;
}

//Inicializacion
let todos = todoReducer()

//Acci√≥n
const newTodo = {
  id:2,
  todo:'Comprar Leche',
  done: false
}

const agregarTodoAction = {
  type:'agregar',
  payload: newTodo
}

todos = todoReducer( todos, agregarTodoAction )

console.log(todos)