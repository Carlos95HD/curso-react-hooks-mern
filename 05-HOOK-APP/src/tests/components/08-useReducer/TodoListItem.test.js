import { shallow } from "enzyme";
import React from "react";
import { TodoListItem } from "../../../components/08-useReducer/TodoListItem";
import { demoTodos } from "../../fixtures/demoTodos";

//demoTodos
describe("Pruebas en <TodoListItem />", () => {
  const handleToggle = jest.fn();
  const handleDelete = jest.fn();

  const wrapper = shallow(
    <TodoListItem
      todo={demoTodos[1]}
      index={1}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );

  test("should mostrar correctamente", () => {
    //Snapshot
    expect(wrapper).toMatchSnapshot();
  });

  test("should llamar la función handleDelete", () => {
    // jest.fn()???
    wrapper.find("button").simulate("click");
    // toHaveBeenCalledWith
    expect(handleDelete).toHaveBeenCalledWith(demoTodos[1].id);
  });

  test("should llamar la función handleToggle", () => {
    wrapper.find("p").simulate("click");
    // toHaveBeenCalledWith
    expect(handleDelete).toHaveBeenCalledWith(demoTodos[1].id);
  });

  test("should mostrar el texto correctamente", () => {
    //contenido del parrafo
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(`2. ${demoTodos[1].desc}`);
  });

  test("should tener la clase complete si el TODO.done === true", () => {
    const todo = demoTodos[0];
    todo.done = true;

    const wrapper = shallow(<TodoListItem todo={todo} />);
    expect(wrapper.find("p").hasClass("complete")).toBe(true);
  });
});
