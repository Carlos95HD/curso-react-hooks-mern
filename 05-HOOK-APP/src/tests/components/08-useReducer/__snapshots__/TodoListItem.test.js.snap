// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoListItem /> should mostrar correctamente 1`] = `
<li
  className="list-group-item"
  key="2"
>
  <p
    className="false"
    onClick={[Function]}
  >
    2
    . 
    Aprender Angular
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
